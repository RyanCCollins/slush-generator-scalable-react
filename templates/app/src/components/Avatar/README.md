## Avatar Component
A reusable avatar component with some styling.

### Example

```js
<Avatar src={person.url} />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **src**    | String   |             | Any string value
