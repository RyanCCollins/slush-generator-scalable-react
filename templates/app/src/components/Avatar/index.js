import React, { PropTypes } from 'react';
import cssModules from 'react-css-modules';
import styles from './index.module.scss';

const defaultAvatarUrl = 'https://github.com/RyanCCollins/cdn/blob/master/alumni-webapp/no-user.png?raw=true';

const Avatar = ({
  src,
}) => (
  <img
    alt="Avatar"
    src={src || defaultAvatarUrl}
    className={styles.avatar}
  />
);

Avatar.propTypes = {
  src: PropTypes.string.isRequired,
};

Avatar.defaultProps = {
  src: defaultAvatarUrl,
};

export default cssModules(Avatar, styles);
