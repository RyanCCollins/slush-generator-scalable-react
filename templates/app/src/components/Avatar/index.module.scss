.avatar {
  width: 100px;
  height: 100px;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.5);
  padding: 4px;
  line-height: 1.42857143;
  background-color: #fff;
  border: 1px solid #ddd;
  border-radius: 50%;
}
