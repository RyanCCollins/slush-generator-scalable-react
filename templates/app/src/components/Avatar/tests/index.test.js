import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import Avatar from '../index';

describe('<Avatar />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <Avatar src="https://github.com/ryanccollins.png" />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
