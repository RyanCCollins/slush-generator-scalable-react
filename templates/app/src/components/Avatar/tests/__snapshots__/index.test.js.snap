exports[`<Avatar /> should render with default props 1`] = `
<img
  alt="Avatar"
  src="https://github.com/ryanccollins.png" />
`;
